Analysis script for Test beam data raw analysis

GNiramay opened the pull request:
This PR will be built upon the current `QIEAnalyzer` to include all the plots related to Test-beam data.

GNiramay committed:
Added Correlation plots, fixed EventReadout functions' return types

bryngemark commented:
If you're going to delete all my histograms, I can't approve this PR 😄 I need them for making my monitoring plots. 

I suggest, if they annoy you, to make them optional with a python configurable argument, or, that we split this off into a separate analyzer. What do you think?

GNiramay commented:
> If you're going to delete all my histograms, I can't approve this PR 😄 I need them for making my monitoring plots. 
> 
> I suggest, if they annoy you, to make them optional with a python configurable argument, or, that we split this off into a separate analyzer. What do you think?

I'm sorry for deleting your histograms. My code kept on crashing for two days, so I deleted everything and started from scratch. We definitely need all those histograms and I'll include them again in the next commit.

My goal with the PR is to include all the important histograms one would need for the analysis. So I'd rather keep them all in one place. We can, of course, discuss which ones to keep.

GNiramay committed:
Adding inter-channel charge correlations and graphs for SiPM noise

GNiramay committed:
Filling the correlation and SiPM noise histograms

GNiramay committed:
Selecting previously selected default graphs

omar-moreno commented:
Please move the analyzer to the analysis repo or put the plots in DQM. This module should be restricted to reconstructions and simulation.

GNiramay commented:
> Please move the analyzer to the analysis repo or put the plots in DQM. This module should be restricted to reconstructions and simulation.

Ok. I talked to others. There is a lot to be done before we could push the code to trunk. Until then we could work with TrigScint branch. When we're ready we could port it to analysis or dqm.

GNiramay committed:
First attempt at fitting fake photon pulses

GNiramay committed:
Debugging work, before ldmx-sw image printing related fix

GNiramay committed:
Guessing initial values of the fit parameters using pulse shape information

GNiramay committed:
Debugging of pulse fitting continues

GNiramay committed:
Single PE pulse fitting successful

tomeichlersmith commented:
With the impending submodule->subdirectory transition on Wednesday, I am commenting here to remind you to either merge this PR before then or close it and [re-apply the changes](https://ldmx-software.github.io/transition-to-ldmx-sw-v4.html) after Wednesday to make a new PR on the subdirectory of ldmx-sw.

tomeichlersmith commented:
I have already done the transition and am in the final stages of testing. [Updating submodule changes to apply to subdirectories](https://ldmx-software.github.io/transition-to-ldmx-sw-v4.html) is not a difficult process and one that I've tested on my own branch of the ECal submodule. Please follow those instructions and re-open this PR in ldmx-sw off the new trunk.
